tracer/src/Datadog.Trace/Configuration/GitMetadataTagsProvider.cs

```diff
@@ -73,7 +73,7 @@ public bool TryExtractGitMetadata([NotNullWhen(true)] out GitMetadata? gitMetada
         catch (Exception e)
         {
             Log.Error(e, "Error while extracting SourceLink information");
-            gitMetadata = GitMetadata.Empty;
+            gitMetadata = _cachedGitTags = GitMetadata.Empty;
```

Comment:
nit - Adding a test making sure we log the error once for instance would make this future proof.

Reply:
Thanks for the feedback! I considered that but I just wasn't sure about the ROI in terms of complexity to accomplish such a test, because:
1. I would need to add some abstractions here to create a `FaultyGitMetadataExtractor` or some such that would trigger an exception, which seems be a bit excessive but probably fine in the context of a unit test, but...
2. I don't know of a way to accomplish this in the context of a unit test, since the only approach to asserting on the contents of our log files is the `LogEntryWatcher`, which I _think_ can only be used in an integration test? I suppose making it an integration test would require somehow passing in an environment variable that would cause us to trigger an exception here, which seems to not just be worth it.

Reply:
I agree generally, it's probably not worth the ROI (certainly not in integration test), but if you want a unit test, I would probably

- Make `GitMetadataTagsProvider.TryGetGitTagsFromSourceLink()` virtual
- Create a derived version that throws when `TryGetGitTagsFromSourceLink()` is called
- Call `TryExtractGitMetadata()` twice
- Assert in `TryGetGitTagsFromSourceLink` that it's only called once

(and red-green test this fix on it)